src/DBAL/Middleware/LaravelDebugbarLogging/ExecutionTime.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace LaravelDoctrine\ORM\DBAL\Middleware\LaravelDebugbarLogging;
+
+use Illuminate\Database\Connection as DatabaseConnection;
+use Illuminate\Database\Events\QueryExecuted;
+
+trait ExecutionTime
+{
+    /**
+     * Measure the execution time of a callable and log it as a query execution event.
+     *
+     * @param  callable   $callable
+     * @param  string     $sql
+     * @param  array|null $params
+     *
+     * @return mixed
+     */
+    protected function time(callable $callable, string $sql, ?array $params = null)
+    {
+        $start = microtime(true);
+        $result = $callable();
+        $end = microtime(true);
+
+        event(new QueryExecuted($sql, $params ?: [], ($end - $start) * 1000, app(DatabaseConnection::class)));
```

Comment:
In fact, there will be a valid connection here (which is specified in the `doctrine.managers.default.connection` config, which in turn is configured in the `database.connections` config.
If there is no `Illuminate\Database\Connection` yet on this step, it will be created with above config.

Reply:
Sounds to me you have `\Illuminate\Database\DatabaseServiceProvider` enabled in config/app.php? 

I do not, and it would be required for this to work, which seems weird.

Reply:
> Sounds to me you have `\Illuminate\Database\DatabaseServiceProvider` enabled in config/app.php?

Yes, this is a standard service provider in Laravel, enabled by default. Would laravel-doctrine even work without it?